spring-kafka/src/main/java/org/springframework/kafka/core/KafkaOperations.java

```diff
@@ -134,6 +135,14 @@
 	ListenableFuture<SendResult<K, V>> send(ProducerRecord<K, V> record);
 
 	/**
+	 * Send the provided {@link ProducerRecord}.
+	 * @param record the record.
+	 * @param callback a {@link Callback} for the send operation.
+	 * @since 1.3
+	 */
+	void send(ProducerRecord<K, V> record, Callback callback);
```

Comment:
Yes, it doesn't make sense to have a `ListenableFuture<?>` as well as a callback.